### ISSUE TITLE
Rule request: add unavailable to unimplemented function

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2127]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

It's common to add unimplemented methods just to fill protocol/init requirements. For example, if you don't use Interface Builder and create an `UIView` subclass overriding init, you'll be asked to implement `init?(coder:)`. Xcode suggests a fix-it that adds an implementation with a `fatalError `.

In addition to the `fatalError`, one can also mark the method with `@available(*, unavailable)` to avoid someone from calling directly. It's just an extra safety net.

I'm not sure if we should trigger a warning for every method that only contains a `fatalError` inside or only the ones with the default template. Also not sure about protocols/overrides.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// would trigger
public required init?(coder aDecoder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
}

// wouldn't trigger
@available(*, unavailable)
public required init?(coder aDecoder: NSCoder) {
     fatalError("init(coder:) has not been implemented")
}

```
3. Should the rule be configurable, if so what parameters should be configurable?

No.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

Probably opt-in.

**Comment by quinntaylor** [comment]
Thanks for implementing this rule! Follow-up issue: https://github.com/realm/SwiftLint/issues/3374
